server.R

```diff
@@ -952,12 +1234,67 @@ shinyServer(function(input, output, session) {
     cat(model()$mtcResults$model$code, fill=FALSE, labels=NULL, append=FALSE)
   })
   
+  output$download_code <- downloadHandler(
+    filename = "code.txt",
+    content = function(file){
+      file.copy("./codes.txt", file)
+    }
+  )
+  
   # 3g-2 Initial values
   output$inits <- renderPrint({
     model()$mtcResults$model$inits
   })
   
-  # 3g-3 Download codes are in the download file.
+  #' Create a download handler for the initial values for a given chain
```

Comment:
Can you edit this function and `create_chain_data_download_handler` so `model()` is an input and then the functions can go back in `downloadbuttons.R`?

Reply:
As shown in the Friday meeting, that's not possible. I tried making the change in MetaInsight in case that was subtly different, but I found that it indeed stopped the download from ever changing when the model was rerun.